chromium: Update to 132.0.6834.83

adalessandro opened the pull request:
chromium: Update to 132.0.6834.83

Release notes:
    https://chromereleases.googleblog.com/2025/01/stable-channel-update-for-desktop_14.html

Build and patch changes:
------------------------------

Build options:
  * Disabled `use_system_libdrm` build option, as no longer supported.
  * https://chromium-review.googlesource.com/c/chromium/src/+/5956068

Added patches:
  * 0013-Revert-Set-Rust-symbol-visibility-to-hidden-when-C-s.patch
  * 0014-pdfium-Fix-missing-PDFiumAPIStringBufferAdapter-temp.patch
  * 0015-Revert-Connect-the-Rust-log-crate-to-the-base-loggin.patch

Updated patches:
  * 0004-Delete-compiler-options-not-available-in-release-ver.patch
  * Rebased remaining patches.

License changes:
------------------------------

Added licenses:
  * third_party/android_deps/libs/com_google_android_gms_play_services_auth_blockstore/LICENSE
  * third_party/android_deps/libs/com_google_android_gms_play_services_fido/LICENSE
  * third_party/android_deps/libs/com_google_android_libraries_identity_googleid_googleid/LICENSE
  * third_party/libtess2/LICENSE
  * third_party/llvm-libc/src/LICENSE.TXT

Removed licenses:
  * third_party/libavifinfo/LICENSE

Updated licenses:
  * third_party/android_deps/libs/com_google_android_gms_play_services_auth/LICENSE
  * third_party/android_deps/libs/com_google_android_gms_play_services_auth_api_phone/LICENSE
  * third_party/android_deps/libs/com_google_android_gms_play_services_auth_base/LICENSE

Test-built (big thanks to @SoumyaWind):
------------------------------

* chromium-ozone-wayland
  * master, clang, MACHINE=qemux86-64

* chromium-x11
  * master, clang, MACHINE=qemux86-64, qemuarm, qemuarm64, raspberrypi4-64

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0009-Adjust-the-Rust-build-to-our-needs.patch:
Why do we not need this patch any more?

adalessandro replied:
@MaxIhlenfeldt my mistake here. Indeed, we need this patch.
* `RUSTC_BOOTSTRAP=1` is needed to enable the nightly features.
  * Thus, despite my note on setting `RUST_CHANNEL = "nightly"` configuration.
* On my testing setup, I didn't catch the case where the toolchain vendor is assumed to be `unknown`. This is definitely something to keep patched.

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0001-Drop-GN-compiler-settings-conflicting-with-OE.patch:
```diff
@@ -110,5 +108,8 @@ index f162751..356a8d4 100644
  config("clang_revision") {
+-- 
+2.47.1
+
```
Please keep the patch file format, i.e.
* don't add a single space to empty lines - I also got them when using `git format-patch`, I just removed them via `sed -i 's/^ $//g' $PATCH_FILE`; and
* drop the signature (after `--`) by passing `--no-signature` to `git format-patch`.

adalessandro replied:
Done, fixed in https://github.com/OSSystems/meta-browser/pull/867/commits/387a102e2d1a92b7f0721138fa5a6303277aa43a

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/chromium-ozone-wayland_132.0.6834.83.bb:
```diff
@@ -20,7 +20,6 @@ GN_ARGS += "\
         use_system_wayland_scanner=true \
         use_xkbcommon=true \
         use_system_minigbm=true \
-        use_system_libdrm=true \
```
Why do we not need this any more?

adalessandro replied:
See https://chromium-review.googlesource.com/c/chromium/src/+/5956068

adalessandro replied:
Updated commit log https://github.com/OSSystems/meta-browser/pull/867/commits/f32f083ee6a9c710adf1a6dd0c0973483b002176 with the related link:

```
 chromium: Disable use_system_libdrm option

A new assertion has been introduced, causing the build to fail with this
option enabled:

| ERROR at //build/config/linux/libdrm/BUILD.gn:18:3: Assertion failed.
|   assert(!is_linux, "System libdrm is not supported in linux")
|   ^-----
| System libdrm is not supported in linux
| See //ui/gfx/BUILD.gn:666:15: which caused the file to be included.
|     deps += [ "//build/config/linux/libdrm" ]

Link: https://chromium-review.googlesource.com/c/chromium/src/+/5956068
Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
```

adalessandro replied:
@MaxIhlenfeldt please review. I think the above explains pretty well the issue and the reason for the change.

MaxIhlenfeldt replied:
Sorry for the delay - yes, this looks good, thanks for the context!

MaxIhlenfeldt reviewed (changes requested):
I haven't reviewed everything yet, but I've left a few comments already.

Regarding the mentioned TODOs:
* you can use this repo's `generate_license_list.py` script to generate the updated license hashes.
* Why do we need to set `"RUST_CHANNEL = "nightly"`? And where would one need to set that?

san-skov-dk commented:
I am trying to build for the i.MX6 cpu and get the following.

80/54278] arm-pmc-linux-gnueabi-clang++ -target arm-pmc-linux-gnueabi   -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -mlittle-endian --dyld-prefix=/usr -Qunused-arguments -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wforma
[81/54278] python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/log_severity.h --depfile obj/base/logging_log_severity_bindgen_generator.d --output gen/base/logging_log_se
FAILED: gen/base/logging_log_severity_bindgen_generator/bindings.rs
python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/log_severity.h --depfile obj/base/logging_log_severity_bindgen_generator.d --output gen/base/logging_log_severity_bind
clang diag: warning: argument unused during compilation: '-mthumb' [-Wunused-command-line-argument]
panicked at bindgen-cli/main.rs:52:36:
Unable to generate bindings: ClangDiagnostic("error: unsupported option '-mfpu=' for target 'x86_64-unknown-linux-gnu'\n")
Traceback (most recent call last):
  File "/thick_disk/san/skov-sdk-v90/build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 118, in <module>
    main()
  File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 106, in main
    subprocess.check_call([args.exe, *genargs], env=env)
  File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/recipe-sysroot-native/usr/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../third_party/rust-toolchain/bin/bindgen', '--no-layout-tests', '--enable-cxx-namespaces', '--no-rustfmt-bindings', '--rust-target', 'nightly', '--depfile', '/thick_disk/san/skov-sdk-v90/buil
[82/54278] python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/rust_log_integration.h --depfile obj/base/logging_rust_log_integration_bindgen_generator.d --output gen/bas
FAILED: gen/base/logging_rust_log_integration_bindgen_generator/bindings.rs
python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/rust_log_integration.h --depfile obj/base/logging_rust_log_integration_bindgen_generator.d --output gen/base/logging_r
clang diag: warning: argument unused during compilation: '-mthumb' [-Wunused-command-line-argument]
panicked at bindgen-cli/main.rs:52:36:
Unable to generate bindings: ClangDiagnostic("error: unsupported option '-mfpu=' for target 'x86_64-unknown-linux-gnu'\n")
Traceback (most recent call last):
  File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 118, in <module>
    main()
  File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 106, in main
    subprocess.check_call([args.exe, *genargs], env=env)
  File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/recipe-sysroot-native/usr/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../third_party/rust-toolchain/bin/bindgen', '--no-layout-tests', '--enable-cxx-namespaces', '--no-rustfmt-bindings', '--rust-target', 'nightly', '--depfile', '/thick_disk/san/skov-sdk-v90/buil
[83/54278] python3 ../../mojo/public/tools/mojom/mojom_parser.py --input-root ../.. --input-root gen --output-root gen --mojom-file-list=gen/cc/mojom/element_id__parser.rsp --check-imports gen/cc/mojom/element_id.build_metadata --enable-

adalessandro commented:
> I am trying to build for the i.MX6 cpu and get the following.
> 
> 80/54278] arm-pmc-linux-gnueabi-clang++ -target arm-pmc-linux-gnueabi -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -mlittle-endian --dyld-prefix=/usr -Qunused-arguments -fstack-protector-strong -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wforma [81/54278] python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/log_severity.h --depfile obj/base/logging_log_severity_bindgen_generator.d --output gen/base/logging_log_se FAILED: gen/base/logging_log_severity_bindgen_generator/bindings.rs python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/log_severity.h --depfile obj/base/logging_log_severity_bindgen_generator.d --output gen/base/logging_log_severity_bind clang diag: warning: argument unused during compilation: '-mthumb' [-Wunused-command-line-argument] panicked at bindgen-cli/main.rs:52:36: Unable to generate bindings: ClangDiagnostic("error: unsupported option '-mfpu=' for target 'x86_64-unknown-linux-gnu'\n") Traceback (most recent call last): File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 118, in main() File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 106, in main subprocess.check_call([args.exe, *genargs], env=env) File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/recipe-sysroot-native/usr/lib/python3.12/subprocess.py", line 413, in check_call raise CalledProcessError(retcode, cmd) subprocess.CalledProcessError: Command '['../../third_party/rust-toolchain/bin/bindgen', '--no-layout-tests', '--enable-cxx-namespaces', '--no-rustfmt-bindings', '--rust-target', 'nightly', '--depfile', 'buil [82/54278] python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/rust_log_integration.h --depfile obj/base/logging_rust_log_integration_bindgen_generator.d --output gen/bas FAILED: gen/base/logging_rust_log_integration_bindgen_generator/bindings.rs python3 ../../build/rust/run_bindgen.py --exe ../../third_party/rust-toolchain/bin/bindgen --header ../../base/logging/rust_log_integration.h --depfile obj/base/logging_rust_log_integration_bindgen_generator.d --output gen/base/logging_r clang diag: warning: argument unused during compilation: '-mthumb' [-Wunused-command-line-argument] panicked at bindgen-cli/main.rs:52:36: Unable to generate bindings: ClangDiagnostic("error: unsupported option '-mfpu=' for target 'x86_64-unknown-linux-gnu'\n") Traceback (most recent call last): File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 118, in main() File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/chromium-132.0.6834.83/out/Release/../../build/rust/run_bindgen.py", line 106, in main subprocess.check_call([args.exe, *genargs], env=env) File "build/tmp/work/cortexa9t2hf-neon-pmc-linux-gnueabi/chromium-ozone-wayland/132.0.6834.83/recipe-sysroot-native/usr/lib/python3.12/subprocess.py", line 413, in check_call raise CalledProcessError(retcode, cmd) subprocess.CalledProcessError: Command '['../../third_party/rust-toolchain/bin/bindgen', '--no-layout-tests', '--enable-cxx-namespaces', '--no-rustfmt-bindings', '--rust-target', 'nightly', '--depfile', '/thick_disk/san/skov-sdk-v90/buil [83/54278] python3 ../../mojo/public/tools/mojom/mojom_parser.py --input-root ../.. --input-root gen --output-root gen --mojom-file-list=gen/cc/mojom/element_id__parser.rsp --check-imports gen/cc/mojom/element_id.build_metadata --enable-

@san-skov-dk could you please retry by pulling the latest changes from this MR?
In any case, more context on your build setup would be great for me to peek/reproduce/help with it. Thanks!

san-skov-dk commented:
@san-skov-dk could you please retry by pulling the latest changes from this MR? In any case, more context on your build setup would be great for me to peek/reproduce/help with it. Thanks!

Still looking into the same issue
Here is my setup for my scarthgap build (of course meta-browser points on the pr)
The serverOS is Alma 9
DISTRO_FEATURES = "argp ext2 largefile usbgadget usbhost xattr nfs zeroconf multiarch systemd alsa opengl wayland usrmerge vulkan rauc security seccomp pam"

[submodule "meta-oe"]
        path = meta-oe
        url = https://git.openembedded.org/meta-openembedded
        branch = scarthgap
[submodule "meta-qt5"]
        path = meta-qt5
        url = https://github.com/meta-qt5/meta-qt5.git
        branch = scarthgap
[submodule "meta-freescale"]
        path = meta-freescale
        url = https://github.com/Freescale/meta-freescale.git
        branch = scarthgap
[submodule "meta-java"]
        path = meta-java
        url = https://git.yoctoproject.org/meta-java
        branch = scarthgap
[submodule "meta-ptx"]
        path = meta-ptx
        url = https://github.com/pengutronix/meta-ptx.git
        branch = scarthgap
[submodule "meta-rauc"]
        path = meta-rauc
        url = https://github.com/rauc/meta-rauc.git
        branch = scarthgap
[submodule "meta-clang"]
        path = meta-clang
        url = https://github.com/kraj/meta-clang.git
        branch = scarthgap
[submodule "meta-browser"]
        path = meta-browser
        url = https://github.com/OSSystems/meta-browser.git
        branch = scarthgap
[submodule "meta-webkit"]
        path = meta-webkit
        url = https://github.com/Igalia/meta-webkit.git
        branch = main
[submodule "meta-security"]
        path = meta-security
        url = https://git.yoctoproject.org/meta-security
        branch = scarthgap
[submodule "poky"]
        path = poky
        url = https://git.yoctoproject.org/poky
        branch = scarthgap
[submodule "meta-lts-mixins"]
        path = meta-lts-mixins
        url = https://git.yoctoproject.org/meta-lts-mixins
        branch = scarthgap/rust
    
If you need more info I will be happy to provide it.

adalessandro commented:
Indeed, this is pretty similar to `0011-Disable-crabbyavif-to-fix-build-errors.patch`.
The issue is related to the way bindgen is configured. To workaround this, I've reverted the
commit the connected bindgen to the rust logger.
Check commit log https://github.com/OSSystems/meta-browser/pull/867/commits/26d3dd9bf3ffcf6ecab94b56fc3a7220b0ea587a

san-skov-dk commented:
> > @san-skov-dk could you please retry by pulling the latest changes from this MR? In any case, more context on your build setup would be great for me to peek/reproduce/help with it. Thanks!

It is building now :)
booth with cortexa9t2hf-neon-pmc-linux-gnueabi and  cortexa53-crypto-pmc-linux

adalessandro commented:
@MaxIhlenfeldt could you do a full PR review?
All the parts should be there now, including license regeneration.
If necessary, of course, we can squash everything up in a single version bump commit.
Let me know if you get the chance to review this, thanks!

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0013-build-config-compiler-BUILD.gn-Disable-CREL.patch:
I think it makes sense to merge this with `0004-Delete-compiler-options-not-available-in-release-ver.patch`.

adalessandro replied:
Yes. Merged. See https://github.com/OSSystems/meta-browser/pull/867/commits/79350e7bfc21ce242757b096f3d95e311c7cee48

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0014-Revert-Set-Rust-symbol-visibility-to-hidden-when-C-s.patch:
```diff
@@ -0,0 +1,57 @@
+Subject: [PATCH] Revert "Set Rust symbol visibility to hidden when C++ symbols
+ are"
+
+This reverts commit ee3900fd57b3c580aefff15c64052904d81b7760.
```
Please also link to https://crrev.com/c/5966273 as that's easier to view.

Also, please mention that we can drop this patch once all our supported configurations have Rust >= 1.83 available (we need https://github.com/rust-lang/rust/pull/130005 and https://github.com/rust-lang/rust/pull/131519). meta-lts-mixins currently has 1.80.1 for kirkstone and 1.81.0 for scarthgap.

adalessandro replied:
Commit log reworked. See https://github.com/OSSystems/meta-browser/pull/867/commits/6a0c61a1e615098ae13d39a9d4347120df010fd4

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0015-pdfium-Fix-missing-PDFiumAPIStringBufferAdapter-temp.patch:
````diff
@@ -0,0 +1,39 @@
+|   173 | using PDFiumAPIStringBufferAdapter =
+|       | ^
+| 1 error generated.
+```
````
I suppose this error doesn't happen with newer versions of clang, which is why upstream doesn't have this? If not too much work (maybe using godbolt.org and a reduced version of the code responsible for the error) it'd be great to find out which version of clang is the first that doesn't throw this error. Then we can mention it in the patch message, and once all supported configurations have that clang version available we can drop the patch.

adalessandro replied:
Testing this chunk of code:
```
#include <string>                                                               

template <class StringType>
class PDFiumAPIStringBufferAdapter {
public:
    PDFiumAPIStringBufferAdapter();
};

template <class StringType>
PDFiumAPIStringBufferAdapter<StringType>::PDFiumAPIStringBufferAdapter()
{}

/* Good */
PDFiumAPIStringBufferAdapter adapter();

/* Bad */
PDFiumAPIStringBufferAdapter<std::u16string> adapter();
```

adalessandro replied:
Can't find a suitable clang version compiling the above. Always getting the following error:
```
<source>:13:1: error: use of class template 'PDFiumAPIStringBufferAdapter' requires template arguments; argument deduction not allowed in function return type
   13 | PDFiumAPIStringBufferAdapter adapter();
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
<source>:4:7: note: template is declared here
    4 | class PDFiumAPIStringBufferAdapter {
      |       ^
1 error generated.
Compiler returned: 1
```

adalessandro replied:
GCC trunk gives the same error:
```
<source>:13:1: error: deduced class type 'PDFiumAPIStringBufferAdapter' in function return type
   13 | PDFiumAPIStringBufferAdapter adapter();
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
<source>:4:7: note: 'template<class StringType> class PDFiumAPIStringBufferAdapter' declared here
    4 | class PDFiumAPIStringBufferAdapter {
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiler returned: 1
```

adalessandro replied:
@MaxIhlenfeldt TBH, I'm not sure on which version this is supposed to work.

MaxIhlenfeldt replied:
The following code works as a test case:

```cpp
#include <string>

namespace internal {

template <class StringType>
class PDFiumAPIStringBufferAdapter {
 public:
  PDFiumAPIStringBufferAdapter(StringType* str) {}
};

}  // namespace internal

template <class StringType>
using PDFiumAPIStringBufferAdapter =
    internal::PDFiumAPIStringBufferAdapter<StringType>;

std::u16string GetPageObjectMarkName() {
  std::u16string name;
  PDFiumAPIStringBufferAdapter adapter(&name);
  return name;
}
```

It compiles with clang 19.1.0, but not with clang 18.1.0 or lower. So please add a note that we can drop the patch once all supported configurations have clang 19 available.

adalessandro replied:
Ah, I understand now the feature, thanks. I've reworked the patch and added version release notes from https://releases.llvm.org/19.1.0/tools/clang/docs/ReleaseNotes.html#c-20-feature-support
Please check https://github.com/OSSystems/meta-browser/compare/1f12168d96918f23b1de6f747c693ebe25ec8eb5..cf7938591ed31f5d7728a45137991c8aff4baf29

MaxIhlenfeldt replied:
That seems to be the exact feature indeed, thanks for finding and documenting that!

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0014-Revert-Set-Rust-symbol-visibility-to-hidden-when-C-s.patch:
````diff
@@ -0,0 +1,57 @@
+[...]
+| error: unknown unstable option: `default-visibility`
+```
+
````
Missing Signed-off-by and Upstream-Status.

adalessandro replied:
See https://github.com/OSSystems/meta-browser/pull/867/commits/72047986a624945b4edd2e3d46fa309dfaf64d92
Added `Upstream-Status: Inappropriate [specific to older versions of rust]`

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0015-pdfium-Fix-missing-PDFiumAPIStringBufferAdapter-temp.patch:
````diff
@@ -0,0 +1,39 @@
+| 1 error generated.
+```
+
+Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
````
Please also add Upstream-Status (probably similar to what we have in e.g. `0012-Revert-Allow-and-use-std-hardware_destructive_interf.patch`).

adalessandro replied:
See https://github.com/OSSystems/meta-browser/pull/867/commits/6b11d8d7098962a4b6f98718f5cf825090b663e9
Added `Upstream-Status: Inappropriate [specific to older versions of clang]`

MaxIhlenfeldt commented on meta-chromium/recipes-browser/chromium/files/0016-Revert-Connect-the-Rust-log-crate-to-the-base-loggin.patch:
```diff
@@ -0,0 +1,488 @@
+
+[0] meta-browser commit fd347dc ("chromium: Update to 127.0.6533.99")
+
+Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
```
Again, please also add Upstream-Status.

MaxIhlenfeldt reviewed (changes requested):
In the previous PRs we also had links to the changes that introduced the license changes (see e.g. #852). @rakuco do you remember why? Maybe we can drop that and just mention which licenses changed/were added/were removed.

rakuco commented:
> In the previous PRs we also had links to the changes that introduced the license changes (see e.g. #852). @rakuco do you remember why? Maybe we can drop that and just mention which licenses changed/were added/were removed.

I think I started doing that ages ago just to make it easier for people actually using the recipe and having to do their due diligence with Chromium's ton of license files and mixes. If it's too much work I think we shouldn't block on this.

rakuco commented:
I wonder if @SoumyaWind would like to do some testing too?

SoumyaWind commented:
@rakuco Sure, I'll start testing and update once I'm done.

MaxIhlenfeldt commented:
> shoud I squash all the commits and have a single "update" commit as https://github.com/OSSystems/meta-browser/commit/0dad36da5970315e53635b4b1c2013e1aec43a29 ? Then add the respective details about patch and license changes?

I think it's enough if you edit the PR description to add the list of added, removed, and updated licenses (no need to link to the CLs introducing these license changes for now I'd say). Then I can let GitHub automatically squash all the commits when merging the PR.

SoumyaWind commented:
For poky scarthgap (with meta-lts-mixns: scarthgap/rust), I encountered the following error during the build:  

>error: failed to run custom build command for `pixbufloader-svg v0.0.1`
>...
>The system library `librsvg-2.0-uninstalled` required by crate `pixbufloader-svg` was not found.
>...
>HINT: you may need to install a package such as librsvg-2.0-uninstalled, librsvg-2.0-uninstalled-dev, or librsvg-2.0-uninstalled-devel.

It seems like **librsvg-2.0-uninstalled.pc** is missing from the **PKG_CONFIG_PATH**. Please let me know if anyone has encountered this error or has suggestions on how to resolve it.

adalessandro commented:
> I have completed testing with the latest changes for poky master. Below are the results for different configurations:
[...]
>     2. **chromium-x11 on qemuarm target**: Encountered the following error during build -
> 
> 
> > arm-poky-linux-gnueabi-ld.lld: error: output file too large: 7086669536 bytes
> > arm-poky-linux-gnueabi-clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > ninja: build stopped: subcommand failed.
> > Updated the configuration to remove the -g flag for ARM. The build was successful for qemuarm and able to verify chromium version using qemuarm emulator

@SoumyaWind I can't reproduce this test case failure. This CI job https://gitlab.collabora.com/mediatek/aiot/bsp/manifest/-/jobs/451917 is building from the current PR on scarthgap for target `qemuarm`. Successfully built `chromium-x11`.

Do you see any difference? Am I missing something in the build test setup?
If you wanna take a look, this is the manifest file from that job: https://gitlab.collabora.com/mediatek/aiot/bsp/manifest/-/blob/f4e85da76e7051589b2bea21f55bb68fa16d1feb/default.xml

adalessandro commented:
> For poky scarthgap (with meta-lts-mixns: scarthgap/rust), I encountered the following error during the build:
> 
> > error: failed to run custom build command for `pixbufloader-svg v0.0.1`
> > ...
> > The system library `librsvg-2.0-uninstalled` required by crate `pixbufloader-svg` was not found.
> > ...
> > HINT: you may need to install a package such as librsvg-2.0-uninstalled, librsvg-2.0-uninstalled-dev, or librsvg-2.0-uninstalled-devel.
> 
> It seems like **librsvg-2.0-uninstalled.pc** is missing from the **PKG_CONFIG_PATH**. Please let me know if anyone has encountered this error or has suggestions on how to resolve it.

@SoumyaWind I have triggered a pipeline with the latest `scarthgap` changes, let's see if we hit the issue there too:
https://gitlab.collabora.com/mediatek/aiot/bsp/manifest/-/pipelines/127818

adalessandro commented:
> > For poky scarthgap (with meta-lts-mixns: scarthgap/rust), I encountered the following error during the build:
> > > error: failed to run custom build command for `pixbufloader-svg v0.0.1`
> > > ...
> > > The system library `librsvg-2.0-uninstalled` required by crate `pixbufloader-svg` was not found.
> > > ...
> > > HINT: you may need to install a package such as librsvg-2.0-uninstalled, librsvg-2.0-uninstalled-dev, or librsvg-2.0-uninstalled-devel.
> > 
> > 
> > It seems like **librsvg-2.0-uninstalled.pc** is missing from the **PKG_CONFIG_PATH**. Please let me know if anyone has encountered this error or has suggestions on how to resolve it.
> 
> @SoumyaWind I have triggered a pipeline with the latest `scarthgap` changes, let's see if we hit the issue there too:
> 
>     * https://gitlab.collabora.com/mediatek/aiot/bsp/manifest/-/pipelines/127819

@SoumyaWind fwiw, the above pipeline succeeded on all targets: qemuarm/qemuarm64/qemux86-64

adalessandro commented:
> In the previous PRs we also had links to the changes that introduced the license changes (see e.g. #852). @rakuco do you remember why? Maybe we can drop that and just mention which licenses changed/were added/were removed.

I've updated the PR description, with patches and licenses changes. Please check.

MaxIhlenfeldt commented:
Could you please also mention that we don't build with `use_system_libdrm=true` any more due to it not being supported since https://chromium-review.googlesource.com/c/chromium/src/+/5956068 (in the "Build and patch changes" section)?

And also please add a "Test-built" section to the PR description, see e.g. #852.

One final question, this PR is only for master, scarthgap and kirkstone will require separate PRs to their respective branches - do you plan to send those once this one has landed? We can discuss the `pixbufloader-svg` error on scarthgap in that PR.

adalessandro commented:
> Could you please also mention that we don't build with `use_system_libdrm=true` any more due to it not being supported since https://chromium-review.googlesource.com/c/chromium/src/+/5956068 (in the "Build and patch changes" section)?

Updated.

> 
> And also please add a "Test-built" section to the PR description, see e.g. #852.

Updated with entries for `master` testing as this PR is only targeting `master` branch.
Note that `qemuarm` not added as, IIUC, still not validated using the current PR due to the error reported by @SoumyaWind .

> 
> One final question, this PR is only for master, scarthgap and kirkstone will require separate PRs to their respective branches - do you plan to send those once this one has landed? We can discuss the `pixbufloader-svg` error on scarthgap in that PR.

Yes, let's make this PR only target `master` branch for the sake of simplicity. And let's have follow up PRs for the scarthgap and kirkstone. Again, note that this PR description only have details for `master` branch in the build test results.

adalessandro commented:
>     2. **chromium-x11 on qemuarm target**: Encountered the following error during build -
> 
> 
> > arm-poky-linux-gnueabi-ld.lld: error: output file too large: 7086669536 bytes
> > arm-poky-linux-gnueabi-clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > ninja: build stopped: subcommand failed.
> > Updated the configuration to remove the -g flag for ARM. The build was successful for qemuarm and able to verify chromium version using qemuarm emulator

@SoumyaWind for the above test case, what was your fix? I haven't had the chance to really reproduce the build issue, so if you have the required patch, could you point that to me? Thanks!

SoumyaWind commented:
> > ```
> > 2. **chromium-x11 on qemuarm target**: Encountered the following error during build -
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > > arm-poky-linux-gnueabi-ld.lld: error: output file too large: 7086669536 bytes
> > > arm-poky-linux-gnueabi-clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > > ninja: build stopped: subcommand failed.
> > > Updated the configuration to remove the -g flag for ARM. The build was successful for qemuarm and able to verify chromium version using qemuarm emulator
> 
> @SoumyaWind for the above test case, what was your fix? I haven't had the chance to really reproduce the build issue, so if you have the required patch, could you point that to me? Thanks!

@adalessandro This is a linker error that has been occurring since Chromium 127.0.6533.99 and is not related to Chromium changes. Refer - https://github.com/OSSystems/meta-browser/pull/828#issuecomment-2273176913.

I have not fixed the issue, but I did not encounter this error in a release build where I explicitly removed debug flags for ARM (CFLAGS:remove:arm = "-g" and CXXFLAGS:remove:arm = "-g").

I have verified the chromium version for qemuarm in this configuration.

kraj commented:
> > > ```
> > > 2. **chromium-x11 on qemuarm target**: Encountered the following error during build -
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > > arm-poky-linux-gnueabi-ld.lld: error: output file too large: 7086669536 bytes
> > > > arm-poky-linux-gnueabi-clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> > > > ninja: build stopped: subcommand failed.
> > > > Updated the configuration to remove the -g flag for ARM. The build was successful for qemuarm and able to verify chromium version using qemuarm emulator
> > 
> > 
> > @SoumyaWind for the above test case, what was your fix? I haven't had the chance to really reproduce the build issue, so if you have the required patch, could you point that to me? Thanks!
> 
> @adalessandro This is a linker error that has been occurring since Chromium 127.0.6533.99 and is not related to Chromium changes. Refer - [#828 (comment)](https://github.com/OSSystems/meta-browser/pull/828#issuecomment-2273176913).
> 
> I have not fixed the issue, but I did not encounter this error in a release build where I explicitly removed debug flags for ARM (CFLAGS:remove:arm = "-g" and CXXFLAGS:remove:arm = "-g").
> 
> I have verified the chromium version for qemuarm in this configuration.

well, it seems objects are getting bigger and more for linker to handle with every release. I have few suggestions which might help

Enable thin LTO (Link Time Optimization) by setting these GN args
```
use_thin_lto = true
thin_lto_enable_optimizations = true
```

Second suggestion
Split the build into multiple parts using the symbol_level GN arg
```
symbol_level = 1 
```
you can use 0 instead of 1 for even smaller output

Third is to use the `is_component_build` flag to create shared libraries instead of one large executable:

```
is_component_build = true

```

maybe if we are lucky one of these tricks will help us

MaxIhlenfeldt reviewed (approved):
I've added qemuarm to the tested list, lgtm now, thanks!
